src/ocrd_network/runtime_data/deployer.py

```diff
@@ -129,41 +129,32 @@ def stop_all(self) -> None:
         self.stop_mongodb()
         self.stop_rabbitmq()
 
-    def start_unix_mets_server(self, mets_path: str) -> Path:
-        log_file = get_mets_server_logging_file_path(mets_path=mets_path)
-        mets_server_url = Path(config.OCRD_NETWORK_SOCKETS_ROOT_DIR, f"{safe_filename(mets_path)}.sock")
+    def start_uds_mets_server(self, ws_dir_path: str) -> Path:
+        log_file = get_mets_server_logging_file_path(mets_path=ws_dir_path)
+        mets_server_url = get_uds_path(ws_dir_path=ws_dir_path)
         if is_mets_server_running(mets_server_url=str(mets_server_url)):
-            self.log.warning(f"The mets server for {mets_path} is already started: {mets_server_url}")
+            self.log.warning(f"The UDS mets server for {ws_dir_path} is already started: {mets_server_url}")
             return mets_server_url
-        cwd = Path(mets_path).parent
         self.log.info(f"Starting UDS mets server: {mets_server_url}")
         sub_process = Popen(
-            args=["nohup", "ocrd", "workspace", "--mets-server-url", f"{mets_server_url}",
-                  "-d", f"{cwd}", "server", "start"],
-            shell=False,
-            stdout=open(file=log_file, mode="w"),
-            stderr=open(file=log_file, mode="a"),
-            cwd=cwd,
-            universal_newlines=True
+            args=["nohup", "ocrd", "workspace", "-U", f"{mets_server_url}", "-d", f"{ws_dir_path}", "server", "start"],
+            stdout=open(file=log_file, mode="w"), stderr=open(file=log_file, mode="a"), cwd=ws_dir_path,
+            shell=False, universal_newlines=True
```

Comment:
Sure, these can be adapted. Thanks for the suggestions.

Reply:
> why nohup and not [Popen(..., start_new_session=True)](https://docs.python.org/3.9/library/subprocess.html#popen-constructor), which IIUC should be equivalent but more portable?

Resolved.

> why subprocess.Popen on the METS Server CLI and not multiprocessing.Process(..., daemon=True) on the METS Server API?

Do you know how to pass stdin and stdout in that case? The following fails:

```sh
        sub_process = multiprocessing.Process(
            args=["ocrd", "workspace", "-U", f"{mets_server_url}", "-d", f"{ws_dir_path}", "server", "start"],
            kwargs={
                "stdout": open(file=log_file, mode="w"),
                "stderr": open(file=log_file, mode="a"),
                "cwd": ws_dir_path,
                "shell": False,
                "universal_newlines": True
            },
            daemon=True
        )
        sub_process.start()
```

Reply:
I don't understand. [Process](https://docs.python.org/3/library/multiprocessing.html#multiprocessing.Process) is supposed to be used with an API (like `OcrdMetsServer.startup`) passed to kwarg `target` not with a CLI.

Reply:
So, if I understand correctly then you ask why the deployer should care about knowing the Mets Server CLI instead of just calling a wrapper and the details are abstracted away inside the Mets Server API?

Reply:
Yes, exactly.

Reply:
Resolved in: https://github.com/OCR-D/core/pull/1220/commits/2b9c184a7fa749b3c554542975f0a0ab9729d79f

Reply:
I'm sorry, I should have read your statement more carefully. I thought you meant the _existing_ OcrdMetsServer API, not some new API merely calling its CLI (as you now implemented).

IMO using `multiprocessing.Process` on the API instead of `subprocess.Popen` on the CLI is better because it is faster, needs no I/O or log handling and no killing – you can just yield to or stop the background thread.

Reply:
The I/O in this case is used for logging the OcrdMetsServer logs to a file. I am not sure why you want to get rid of that. The class OcrdMetsServer is instantiated separately for every workspace. Do you mean a single OcrdMetsServer process but many threads for the different instantiations? I am a bit confused on that one.

Reply:
I did not mean actual threads, I am still referring to `multiprocessing.Process`. One background process per METS Server / workspace – but as `Process` object (API) instead of a `Popen` object (CLI).

I thought the logs should go to the processing job log file as well. But I can now see it would take additional configuration to have the forked subprocess log to the corresponding process job log file.

If you don't see benefits in avoiding additional CLI parsing, then let's leave it at that – it's a cosmetic point anyway.

Reply:
> One background process per METS Server / workspace – but as Process object (API) instead of a Popen object (CLI).

Yeah, but then we lose the separate logging per mets file, that's why I asked if you know a workaround for that.

> I thought the logs should go to the processing job log file as well.

Only OCR-D processor logs go inside the processing job logs. The mets server logs just logs the requests and responses for debugging purposes if needed. 

> If you don't see benefits in avoiding additional CLI parsing, then let's leave it at that – it's a cosmetic point anyway.

I just do not see how the conversion could be done while keeping the separate mets server log files. The logging has caused some big issues in the past. The separation of logging is more straightforward with the current implementation. If it's just a cosmetic point, then I would just avoid making changes on that part and potentially mess the logging.